### DISCUSSION TITLE
Using Aux Ports for tool length sensor

### DISCUSSION THREAD

**Discussion by Switch-Witch** [grblHAL/core#566] [Q&A]
Hi,
I connected my edge finder to the probe port and now I asked myself if it is possible to connect the tool lenght sensor to one of the aux ports. The question is: Is it possible and where to define the aux port pin as a second probe port?

**Comment by karoria** [comment]
You don't need separate aux port for tool height sensing if both the probe and tool height sensor are of same voltage circuit. For NC type wiring you can series connect both and for NO type wiring, you can connect them parallelly. Your rest of the job is to write different macros for probe and tool height sensing using G38.2. As both the task will never run simultaneously, it will not have any issues. I hope I was able to communicate clearly.
Cheers!
Ravi

**Reply by Switch-Witch** [reply]
Hi Ravi,
thx a lot for your answer! I know that running them parallel would be an option and the go to solution. The question was more like: would it be possible... 
Running some printers with klipper something like using a port with a dedicated pin for another usage is quite usual and I am interested in knowing if something like this would be possible in grblhal.
cheers
Kai

**Reply by terjeio** [reply]
Use a gpio (aux out) controlled relay and [this](https://github.com/grblHAL/Templates/blob/master/my_plugin/Probe_select/my_plugin.c) or [this](https://github.com/grblHAL/Templates/blob/master/my_plugin/Probe_select2/my_plugin.c) plugin. Can be added from the Web Builder _3rd party plugins_ tab or by [copying into a locally compiled project](https://github.com/grblHAL/Templates/blob/master/my_plugin/README.md).

**Reply by Switch-Witch** [reply]
Thx a lot for your answer! 😘
